docs/book/src/user/quick-start.md

```diff
@@ -573,6 +573,7 @@ spec:
       apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
       kind: KubeadmConfig
       name: capi-quickstart-controlplane-0
+      namespace: default
```

Comment:
Isn't `default` assumed when the namespace is empty? Were you able to to `kubectl get` this object when the controller returned the error in the linked issue?

Reply:
It will use whatever namespace is listed in the current kubeconfig context, and I guess if one's not set, then it would fall back to default.

I'm not sure this change is necessary?

Reply:
We also have a defaulting webhook that sets the namespace to match the machine's namespace, and a validating webhook to reject if they don't match.

Reply:
> It will use whatever namespace is listed in the current kubeconfig context

Which client are you talking about? Kubectl, controller-runtime, or both?

Reply:
Generally anything using client-go to get a `RESTConfig` using the most common methods to do so.

Reply:
@vincepri @ncdc  yes I was able to see the KubeadmConfig object when the controller returned the error as described in https://github.com/kubernetes-sigs/cluster-api/issues/2403 . That is what confused me as I expected it to pick the default namespace. I can still reproduce this issue with v1alpha2. Log snippet is present in the original issue.

Reply:
Could you please gist/snippet the full logs from capi, cabpk, and capv? Thanks!

Reply:
Shared on #2403. Thanks for looking into it.